mozilla-release/browser/components/sessionstore/SessionStore.jsm

```diff
@@ -2961,19 +2961,44 @@ var SessionStoreInternal = {
     // This is intended to work when both tab restore and fresh tab features
     // are enabled.
     if (!overwriteTabs) {
+      const uriFixup = Services.uriFixup;
+      function fixupURI(pageAddress) {
+        try {
+          return uriFixup.createFixupURI(pageAddress, uriFixup.FIXUP_FLAG_NONE);
```

Comment:
Not the same. The purpose here is to fix user input, not parse a well-formed URI. To feel the difference, run this in dev console:
`Services.io.newURI('cliqz.com', null, null);`